Libraries/Drei/Tests/testing.lisp

```diff
@@ -101,6 +101,10 @@ display. Used for testing.")
            (with-bound-drei-special-variables (,drei :minibuffer nil)
              ,@body))))))
 
+(defmethod climb:publish-selection (port selection object object-type)
+  "Empty implementation to allow with-drei-environment to contain editing operations."
+  (declare (ignore port selection object object-type)))
```

Comment:
this should be at all times specialized on the first argument (at least). you may use eql-specializer for instance (see the bottom of file Tests/tests.lisp). given it is not used at all here maybe just remove the method from PR. other things look good to me.

Reply:
> given it is not used at all here maybe just remove the method from PR. other things look good to me.

That function is called from all tests that edit a buffer, I'll paste from the first failed test:

- Without having loaded `mcclim`:
```
 BACKWARD-KILL-PARAGRAPH []: 
      Unexpected Error: #<UNDEFINED-FUNCTION PUBLISH-SELECTION {100456A493}>
The function CLIM-EXTENSIONS:PUBLISH-SELECTION is undefined...
```

- After having loaded `mcclim`:
```
 BACKWARD-KILL-PARAGRAPH []: 
      Unexpected Error: #<SB-PCL::NO-APPLICABLE-METHOD-ERROR {100A3C9503}>
There is no applicable method for the generic function
  #<STANDARD-GENERIC-FUNCTION CLIM-EXTENSIONS:PUBLISH-SELECTION (4)>
when called with arguments
  (NIL :CLIPBOARD "This is the last paragraph." STRING).
See also:
  The ANSI Standard, Section 7.6.6..
```

Reply:
then it should be fixed in drei to not call publish-selection if there is no sheet. fix will probably look like this (in kill-ring):

```
(defun push-start-and-copy (chain vector)
  (push-start chain vector)
  (when (every #'characterp vector)
    (when-let ((pane (drei:editor-pane (drei:drei-instance))))
      (clime:publish-selection pane :clipboard (coerce vector 'string) 'string))))
```

(previously we did not call when-let but only let)

Reply:
Okay, I made the change.